src/reducers/select.js

```diff
@@ -0,0 +1,134 @@
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*
+* @licend  The above is the entire license notice
+* for the JavaScript code in this file.
+*
+*/
+//import createDebugLogger from 'debug';
+import {normalizeSync} from 'normalize-diacritics';
+
+export function strictEquality(subfieldA, subfieldB) {
+  return subfieldA.code === subfieldB.code &&
+  subfieldA.value === subfieldB.value;
+}
+
+export function subsetEquality(subfieldA, subfieldB) {
+  return subfieldA.code === subfieldB.code &&
+  (subfieldA.value.indexOf(subfieldB.value) !== -1 || subfieldB.value.indexOf(subfieldA.value) !== -1);
+}
+
+export default ({pattern, equalityFunction = strictEquality}) => (base, source) => {
+  //const debug = createDebugLogger('@natlibfi/marc-record-merge');
+  const baseFields = base.get(pattern);
+  const sourceFields = source.get(pattern);
+  return selectFields();
+
+  function selectFields() {
+    checkFieldType(baseFields);
+    checkFieldType(sourceFields);
+
+    if (baseFields.length > 1 || sourceFields.length > 1) {
+      return base;
+    }
+    const [baseField] = baseFields;
+    const [sourceField] = sourceFields;
+
+    if (baseField.tag === sourceField.tag === false) {
+      return base;
+    }
+    const baseSubs = baseField.subfields;
+    const sourceSubs = sourceField.subfields;
+
+    const baseSubsNormalized = baseSubs
+      .map(({code, value}) => ({code, value: normalizeSubfieldValue(value)}));
+
+    const sourceSubsNormalized = sourceSubs
+      .map(({code, value}) => ({code, value: normalizeSubfieldValue(value)}));
+
+    const equalSubfieldsBase = baseSubsNormalized
+      .filter(baseSubfield => sourceSubsNormalized
+        .some(sourceSubfield => equalityFunction(baseSubfield, sourceSubfield)));
+
+    const equalSubfieldsSource = sourceSubsNormalized
+      .filter(sourceSubfield => baseSubsNormalized
+        .some(baseSubfield => equalityFunction(sourceSubfield, baseSubfield)));
+
+    const equalSubsetsBase = baseSubsNormalized
+      .filter(baseSubfield => sourceSubsNormalized
+        .some(sourceSubfield => equalityFunction(baseSubfield, sourceSubfield)));
+
+    const equalSubsetsSource = sourceSubsNormalized
+      .filter(sourceSubfield => baseSubsNormalized
+        .some(baseSubfield => equalityFunction(sourceSubfield, baseSubfield)));
+
+    if (baseSubs.length === sourceSubs.length && equalSubfieldsBase.length < baseSubs.length) {
+      return base;
+    }
+
+    if (baseSubs.length === sourceSubs.length && equalSubfieldsBase.length === equalSubfieldsSource.length) {
+      const totalSubfieldLengthBase = baseSubsNormalized
+        .map(({value}) => value.length)
+        .reduce((acc, value) => acc + value);
+      const totalSubfieldLengthSource = sourceSubsNormalized
+        .map(({value}) => value.length)
+        .reduce((acc, value) => acc + value);
+
+      if (totalSubfieldLengthBase >= totalSubfieldLengthSource) {
+        return base;
+      }
```

Comment:
Unnecessary. You should always return from conditional blocks, so the default should be to return base.